docs/distributions/nixos/installation.md

```diff
@@ -185,15 +209,78 @@ If you would like to organize your configuration a little better, check out othe
 
 ## Wi-Fi and Bluetooth setup
 
+### Pre-Installation Steps
+
+Choose a method below and carefully read the notes. Afterwards, follow the [Wi-Fi and Bluetooth Guide on macOS](../../guides/wifi-bluetooth.md#setting-up) (firmware guide) and come back.
+
+The methods outlined below corresponds to the ones listed in the guide.
+
+=== ":fontawesome-solid-check: Method 1"
+    Method 1 requires you to run the firmware script twice, both in macOS and Linux.
+
+    While this method is easier to set up than Method 2, it does not necessarily mean it is simpler to maintain.
+
+=== ":fontawesome-solid-star: Method 2"
+    This method creates a tarball with the renamed firmware files on macOS. No scripts will need to be run on Linux. This method is more robust than Method 1, but requires some manual configuration.
+
+    This is the recommended method to follow on NixOS.
+
+=== ":fontawesome-solid-ban: Method 3"
+    !!! danger
+        This method is not supported on NixOS as the distribution does not use a conventional package format. A similar setup can be done by following the [Declarative Setup](#declarative-setup) section, but you will need to follow some other method first.
+
+=== ":fontawesome-solid-hourglass: Method 4"
+    <!-- TODO: after testing APFS driver and inclusion in the ISO, update this section. -->
+    !!! warning
+        This method is not **yet** supported, please stay tuned. In the meantime, please use other methods.
+
+=== ":fontawesome-solid-check: Method 5"
+    This method is similar to Method 1, but instead of requiring macOS to be installed, the firmware files are extracted from recovery images from Apple directly. This is useful if you have removed macOS or the partition is not accessible for any reason.
```

Comment:
the firmware files are extracted from recovery images downloaded directly from Apple.